net-misc/kea/kea-3.0.0.ebuild

```
@@ -0,0 +1,282 @@
+	>=dev-build/meson-1.8
+	doc? (
+		$(python_gen_any_dep '
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]
+		')
+	)
+	virtual/pkgconfig
+"
+
+python_check_deps() {
+	use doc || return 0;
+	python_has_version "dev-python/sphinx[${PYTHON_USEDEP}]" \
+		"dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	if use doc || use shell; then
+		python_setup
+	fi
+}
+
+src_prepare() {
+	default
+
+	# Remove documentation from build
+	if use !doc; then
+		sed -e "/^subdir('doc')$/d" \
+		-i meson.build || die
+	fi
+
+	# Fix up doc path where examples go
+	sed -e "s:DATADIR / 'doc/kea':DATADIR / 'doc/${P}':" \
+		-i meson.build || die
+
+	# set shebang before meson if we are installing the shell
+	if use shell; then
+		sed -e 's:^#!@PYTHON@:#!/usr/bin/env python3:' \
+		-i src/bin/shell/kea-shell.in || die
+	fi
+
+	# Don't allow meson to install shell
+	sed -e 's:install\: true:install\: false:' \
+		-i src/bin/shell/meson.build || die
+
+	# do not create /run
+	sed -e '/^install_emptydir(RUNSTATEDIR)$/d' \
+		-i meson.build || die
+}
+
+src_configure() {
+	# https://bugs.gentoo.org/861617
+	# https://gitlab.isc.org/isc-projects/kea/-/issues/3946
+	#
+	# Kea Devs say no to LTO
+	filter-lto
+
+	local emesonargs=(
+		--localstatedir="${EPREFIX}/var"
+		-Drunstatedir="${EPREFIX}/run"
+		-Dkrb5=disabled
+		-Dnetconf=disabled
+		$(meson_feature mysql)
+		$(meson_feature postgres postgresql)
+		$(meson_feature test tests)
+	)
+	if use debug; then
+		emesonargs+=(
+			--debug
+		)
+	fi
+	if use openssl; then
+		emesonargs+=(
+			-Dcrypto=openssl
+		)
+	else
+		emesonargs+=(
+			-Dcrypto=botan
+		)
+	fi
+
+	meson_src_configure
+}
+
+src_test() {
+
+	# Get list of all tests
+	pushd "${BUILD_DIR}" || die
+	local ALL_TESTS=$(meson test --list | cut -d / -f 2) || die
+	popd
+
+	# Some tests will fail due to sandbox restrictions, heres an array of those tests
+	local SKIP_TESTS=(
+		dhcp-radius-tests
+		kea-log-buffer_logger_test.sh
+		kea-log-console_test.sh
+		dhcp-lease-query-tests
+		dhcp4_process_tests.sh
+		dhcp6_process_tests.sh
+		kea-dhcp6-tests
+		d2_process_tests.sh
+		ca_process_tests.sh
+		admin_tests.sh
+		memfile_tests.sh
+		keactrl_tests.sh
+		ca_basic_auth_tests.sh
+		d2_basic_auth_tests.sh
+		dhcp4_basic_auth_tests.sh
+		dhcp6_basic_auth_tests.sh
+		shell_ca_process_tests.sh
+		shell_d2_process_tests.sh
+		shell_dhcp4_process_tests.sh
+		shell_dhcp6_process_tests.sh
+		tls_ca_process_tests.sh
+		tls_d2_process_tests.sh
+		tls_dhcp4_process_tests.sh
+		tls_dhcp6_process_tests.sh
+	)
+
+	if [ $(tc-get-ptr-size) -eq 4 ]; then
+		# see https://bugs.gentoo.org/958171 for reason for skipping these tests
+		SKIP_TESTS+=(
+			kea-util-tests
+			kea-dhcp-tests
+			kea-dhcp4-tests
+			kea-dhcpsrv-tests
+			dhcp-ha-lib-tests
+			kea-d2-tests
+		)
+	fi
+
+	local FINAL_TESTS=
+
+	# Create a Final list of tests by removing any matching tests in SKIP_TESTS from ALL_TESTS
+	for CANDIDATE in $ALL_TESTS; do
```

We're in bash and we have arrays, so:
```suggestion
	for CANDIDATE in "${ALL_TESTS[@]}"; do
```